### ISSUE TITLE
[QUESTION] Chapter 2: Definition of `similarities` is subject to information leakage?

### ISSUE THREAD

**Issue by liganega** [ageron/handson-ml3#125]
This question is referring to [the jupyter notebook of Chapter 2](https://github.com/ageron/handson-ml3/blob/main/02_end_to_end_machine_learning_project.ipynb).

===

The code below creates new 10 similarity features based on the location of the districts.
But it also uses the information of `"median_house_value"` as sample weight.

```python
housing_labels = strat_train_set["median_house_value"].copy()
...
similarities = cluster_simil.fit_transform(housing[["latitude", "longitude"]],
                                           sample_weight=housing_labels)
```

But isn't it kind of information leakage to the model? 
The model is going to be trained on predicting the median house value and should NOT have any direct information about it.

**Comment by liganega** [comment]
The `"median_income"` feature would be adequate for sample weight.

**Comment by liganega** [comment]
It is even more questionable whether this preprocessing of clustering could be used because the clustering process would run differently depending on the input dataset. 

The clustering info would be different for two districts when one of them belongs to the training set and the other to test set even though they are similar. 

Am I missing something?

**Comment by liganega** [comment]
In fact, the `sample_weight` option is used only for the demonstration of how to use `ClusterSimilarity` class and is ignored after that. There is therefore no information leakage during the training. 

However, it is still misleading to use `"median_house_value"` as the value for `sample_weight`.  Using instead `"median_income"` results in almost the same clustering.
